server/shared/storage/proxy/mw.js

```diff
@@ -7,6 +7,13 @@ const path = require('path');
 const router = require('express').Router();
 const psl = require('psl');
 
+const IPV4_REGEX = /^((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])(\.(?!$)|$)){4}$/;
+
+function getDomain() {
+  if (IPV4_REGEX.test(config.hostname)) return null;
+  return psl.parse(config.hostname).domain;
+}
```

Comment:
Does this work only for the local proxy provider?

Reply:
Mostly, as domain is not specified and thus determined by viewer URL, which in turn won't expose signed cookies to CloudFront. 
There is possibility for this to work with CloudFront in following circumstances, host name is set to IPv4, viewer URL is different from host name and CloudFront CNAME option is set to match viewer URL, thus signed cookies are set on correct domain, but I doubt this as a real use case.
This fix mostly targets Local proxy, as without it local proxy cannot work when host name is IPv4.

Reply:
@droguljic got it. Can we throw an error in case that the proxy is set to CloudFront and the hostname is IPV4 to warn developer that their configuration is invalid?

Reply:
@MiroDojkic done. Please see the latest commit.

Reply:
@MiroDojkic That was my initial plan, but host name is not available inside storage provider. Importing `config` inside storage provider didn't seem right, so I opted for proxy middleware as config is already available there.
Maybe I missed something, as I'm not that familiar with the codebase, so feel free to provide suggestion.

Reply:
@droguljic since this is basically config validation, it seems to me that we could perform it inside the root server config file.

Reply:
@MiroDojkic inside this file https://github.com/ExtensionEngine/tailor/blob/release/5.0/config/server/index.js?
In order to put it here, logic for testing is something IPv4 needs to be extracted into separate function, probably into `server/shared/util/isIpv4`. Importing something from `server` into `config` seems a bit weird and copy pasting regular expression is not DRY, so I'm in dilemma again.

Reply:
@droguljic yep. If it's not being used anywhere else, we could put it at the bottom of `config/server/index.js`. If the file becomes too bloated with noise, would it make sense to extract it to something like `config/server/validation.js`?

Reply:
@MiroDojkic it needs to be used in two places, `server/shared/storage/proxy/mw.js` and `config/server/index.js`, already is used in former and will be added to latter. Maybe adding new dependency [is-ip](https://www.npmjs.com/package/is-ip) is a better option and it'll definitely resolve this dilemma, so going with new dependency.

Reply:
@droguljic good idea!

Reply:
@MiroDojkic done.